test/end-to-end/offer.js

```diff
@@ -1103,4 +1109,400 @@ describe("Offer endpoint tests", () => {
+                });
+            });
+
+            describe("testing dates editing", () => {
+                test("should fail if publishDate after offer publishEndDate", async () => {
+                    const res = await test_agent
+                        .post(`/offers/edit/${future_test_offer._id.toString()}`)
+                        .send(withGodToken({
+                            "publishDate":
+                            (new Date(new Date(future_test_offer.publishEndDate).getTime() + DAY_TO_MS))
+                                .toISOString()
+                        }))
+                        .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+                    expect(res.body.errors[0]).toHaveProperty("msg", `Error: ${ValidationReasons.MUST_BE_BEFORE("publishEndDate")}`);
+                });
+
+                test("should fail if publishEndDate before offer publishDate", async () => {
+                    const res = await test_agent
+                        .post(`/offers/edit/${future_test_offer._id.toString()}`)
+                        .send(withGodToken({
+                            "publishEndDate":
+                            (new Date(new Date(future_test_offer.publishDate).getTime() - DAY_TO_MS))
+                                .toISOString()
+                        }))
+                        .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+                    expect(res.body.errors[0]).toHaveProperty("msg", `Error: ${ValidationReasons.MUST_BE_AFTER("publishDate")}`);
+                });
+
+                test("should fail if sending invalid date combination in request", async () => {
+                    const res = await test_agent
+                        .post(`/offers/edit/${future_test_offer._id.toString()}`)
+                        .send(withGodToken({
+                            "publishEndDate": new Date(Date.now() + DAY_TO_MS).toISOString(),
+                            "publishDate": new Date(Date.now() + (2 * DAY_TO_MS)).toISOString(),
+                        }))
+                        .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+                    expect(res.body.errors[0]).toHaveProperty("msg", `Error: ${ValidationReasons.MUST_BE_AFTER("publishDate")}`);
+                });
+
+                test("should edit if is after offer's publishEndDate", async () => {
+                    const newDate = (new Date(new Date(future_test_offer.publishEndDate).getTime() - DAY_TO_MS));
+                    const res = await test_agent
+                        .post(`/offers/edit/${future_test_offer._id.toString()}`)
+                        .send(withGodToken({ "publishDate": newDate.toISOString() }))
+                        .expect(HTTPStatus.OK);
+                    expect(res.body).toHaveProperty("publishDate", newDate.toISOString());
+                });
+
+                test("should edit if is after offer's publishEndDate", async () => {
```

Comment:
These two tests have the same name, please differentiate them (it's not a problem if the title is long, as long as it's understandable what the test is testing)